components/TARGET_PSA/services/crypto/COMPONENT_PSA_SRV_IPC/psa_crypto_spm.c

```diff
@@ -689,22 +696,18 @@ psa_status_t psa_asymmetric_decrypt(psa_key_slot_t key,
 /* PSA_KEY_MANAGMENT */
 /****************************************************************/
 
-psa_status_t psa_get_key_lifetime(psa_key_slot_t key,
-                                  psa_key_lifetime_t *lifetime)
+psa_status_t psa_allocate_key(psa_key_handle_t *key_handle)
 {
     psa_error_t err_call;
-    psa_key_mng_ipc_t psa_key_mng_ipc = { 0, 0, 0 };
-    psa_key_mng_ipc.key = key;
-    psa_key_mng_ipc.type = 0;
-    psa_key_mng_ipc.func = PSA_GET_KEY_LIFETIME;
+    psa_key_mng_ipc_t psa_key_mng_ipc = { 0, 0, 0, 0 };
+    psa_key_mng_ipc.handle = *key_handle;
+    psa_key_mng_ipc.func = PSA_ALLOCATE_KEY;
     psa_handle_t handle = PSA_NULL_HANDLE;
     psa_invec_t in_vec[1] = {
-        { &psa_key_mng_ipc, sizeof(psa_key_mng_ipc_t) }
+        { &psa_key_mng_ipc, sizeof(psa_key_mng_ipc) }
     };
-    psa_outvec_t out_vec[1] = { {
-            lifetime, (lifetime == NULL ? 0 :
-                       sizeof(*lifetime))
-        }
+    psa_outvec_t out_vec[1] = {
+        { key_handle, sizeof(*key_handle) }
```

Comment:
can it return null?

Reply:
if the allocate fails then `key_handle` remains untouched